mysticeti-core/src/network.rs

```diff
@@ -113,15 +119,28 @@ impl Network {
                 .run(receiver),
             );
         }
-        handle.spawn(
+        let (stop, rx_stop) = tokio::sync::mpsc::channel(1);
+        let server_handle = handle.spawn(async {
             Server {
                 server,
                 worker_senders,
             }
-            .run(),
-        );
+            .run(rx_stop)
+            .await
+        });
         Self {
             connection_receiver,
+            stop: Some(stop),
+            server_handle: Some(server_handle),
+        }
+    }
+
+    pub async fn shutdown(mut self) {
+        if let Some(stop) = self.stop.take() {
+            let _ = stop.send(()).await;
```

Comment:
nit: I usually use `ok()` instead of `let _ =` pattern, since `let _ =` can ignore both futures and errors, and in this past this has already lead to issues

Reply:
Nice trick, I'll use that!